### PR TITLE
Migrate CI to use Github Actions

### PR DESCRIPTION
**Pull request by justinchuby**
Migrate CI to use Github Actions. Update Google Test to a newer version (some version in 2022). The latest commit has errors I don't know how to fix.

Use `FetchContent` instead of `ExternalProject` to bring in googletest according to its documentation https://google.github.io/googletest/quickstart-cmake.html

Fixes https://github.com/onnx/onnx/issues/6074 Fixes https://github.com/onnx/onnx/issues/6073

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,118 @@
+name: CI
```
Comment (gramalingam) [inline .github/workflows/main.yml:1]: The azure version has a filename and top-level name that mentions "linux ci" ... is that useful to capture? Or hard?

Comment (justinchuby) [reply .github/workflows/main.yml:1]: I am going to merge all three files to one. Starting with linux. Once they are merged the platform names will display in the runs below.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Minimum CMake required
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.14)
```
Comment (justinchuby) [inline CMakeLists.txt:2]: Needed for cmake FetchContent

**Review by gramalingam — APPROVED** [review]

**Comment by gramalingam** [comment]
Will try closing and reopening this PR

**Pull request merged**